components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityUtil.java

```diff
@@ -1434,16 +1435,18 @@ public static int getDefaultItemsPerPage() {
     public static boolean isGroupsVsRolesSeparationImprovementsEnabled() {
 
         try {
-            if (AdminServicesUtil.getUserRealm() == null) {
+            UserRealm userRealm = AdminServicesUtil.getUserRealm();
+            if (userRealm == null) {
                 log.warn("Unable to find the user realm, thus GroupAndRoleSeparationEnabled is set as FALSE.");
                 return Boolean.FALSE;
             }
-            diagnosticLog.info("'GroupAndRoleSeparationEnabled' property is set to: " +
-                    UserCoreUtil.isGroupsVsRolesSeparationImprovementsEnabled(AdminServicesUtil.getUserRealm()
-                    .getRealmConfiguration()));
-            return UserCoreUtil.isGroupsVsRolesSeparationImprovementsEnabled(AdminServicesUtil.getUserRealm()
-                    .getRealmConfiguration());
-        } catch (org.wso2.carbon.user.core.UserStoreException | CarbonException e) {
+            if (groupsVsRolesSeparationImprovementsEnabled == null) {
+                groupsVsRolesSeparationImprovementsEnabled = UserCoreUtil.isGroupsVsRolesSeparationImprovementsEnabled(
+                        userRealm.getRealmConfiguration());
+            }
+            return groupsVsRolesSeparationImprovementsEnabled;
+
+        } catch (UserStoreException | CarbonException e) {
```

Comment:
Any reason to change the UserStoreException class that has been used before?

Reply:
Because for the userRealm variable, org.wso2.carbon.user.api.UserRealm is used (UserCoreUtil.isGroupsVsRolesSeparationImprovementsEnabled method signature paramter also using  org.wso2.carbon.user.api.UserRealm)